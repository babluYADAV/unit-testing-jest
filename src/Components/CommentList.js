import React from "react";

const CommentList = ({ allCommnents }) => {
  if (allCommnents.length == 0) {
    return <h2>No commnets available</h2>;
  }
  return (
    <div>
      <ul>
        {allCommnents.map(({ id, comment }) => {
          return <li key={id}>{comment}</li>;
        })}
      </ul>
    </div>
  );
};

export default CommentList;
