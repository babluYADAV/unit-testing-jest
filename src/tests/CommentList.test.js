import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import CommentList from "../Components/CommentList";

describe("Comment list test", () => {
  const comments = [
    { id: 1, comment: "comment 1" },
    { id: 2, comment: "comment 2" },
  ];
  test("snapshot with no items", () => {
    const tree = renderer.create(<CommentList allCommnents={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("snapshot with items", () => {
    const tree = renderer
      .create(<CommentList allCommnents={comments} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("initial condtions or no comments available", () => {
    render(<CommentList allCommnents={[]} />);
    const h2Element = screen.getByText("No commnets available", {
      exact: false,
    });
    expect(h2Element).toBeInTheDocument();
  });

  test("comments available", () => {
    render(<CommentList allCommnents={comments} />);
    const h2Element = screen.queryByText("No commnets available", {
      exact: false,
    });
    expect(h2Element).not.toBeInTheDocument();
    const commentLi = screen.getAllByRole("listitem");
    expect(commentLi.length).toBe(comments.length);
  });
});
