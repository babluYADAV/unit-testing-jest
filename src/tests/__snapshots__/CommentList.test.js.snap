// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment list test snapshot with items 1`] = `
<div>
  <ul>
    <li>
      comment 1
    </li>
    <li>
      comment 2
    </li>
  </ul>
</div>
`;

exports[`Comment list test snapshot with no items 1`] = `
<h2>
  No commnets available
</h2>
`;
